src/SldStyleParser.ts

```diff
@@ -510,404 +518,463 @@ export class SldStyleParser implements StyleParser<string> {
+          case '#text':
+            return labelEl['#text'];
+          case 'Literal':
+            return labelEl?.[labelName]?.[0 ]?.['#text'] || labelEl?.[labelName]?.[0]?.['#cdata']?.[0]?.['#text'];
+          case 'PropertyName':
+            const propName = labelEl[labelName][0]['#text'];
+            return `{{${propName}}}`;
+          default:
+            return '';
+        }
+      })
+      .join('');
+    return label;
+  };
 
-      pointSymbolizer = this.getIconSymbolizerFromSldSymbolizer(sldSymbolizer);
+  /**
+   * Get the geostyler-style FillSymbolizer from a SLD Symbolizer.
+   *
+   * PolygonSymbolizer Stroke is just partially supported.
+   *
+   * @param sldSymbolizer The SLD Symbolizer
+   * @return The geostyler-style FillSymbolizer
+   */
+  getFillSymbolizerFromSldSymbolizer(sldSymbolizer: any): GsFillSymbolizer {
+    const fillSymbolizer: GsFillSymbolizer = {
+      kind: 'Fill'
+    };
+    const strokeEl = get(sldSymbolizer, 'Stroke');
+    const fillEl = get(sldSymbolizer, 'Fill');
+
+    const fillOpacity = getParameterValue(fillEl, 'fill-opacity', this.sldVersion);
+    const color = getParameterValue(fillEl, 'fill', this.sldVersion);
+
+    const outlineColor = getParameterValue(strokeEl, 'stroke', this.sldVersion);
+    const outlineWidth = getParameterValue(strokeEl, 'stroke-width', this.sldVersion);
+    const outlineOpacity = getParameterValue(strokeEl, 'stroke-opacity', this.sldVersion);
+    const outlineDashArray = getParameterValue(strokeEl, 'stroke-dasharray', this.sldVersion);
+    // const outlineDashOffset = getParameterValue(strokeEl, 'stroke-dashoffset', this.sldVersion);
+
+    const graphicFill = get(sldSymbolizer, 'Fill.GraphicFill');
+    if (graphicFill) {
+      fillSymbolizer.graphicFill = this.getPointSymbolizerFromSldSymbolizer(
+        graphicFill
+      );
+    }
+    if (color) {
+      fillSymbolizer.color = color;
+    }
+    if (fillOpacity) {
+      fillSymbolizer.fillOpacity = Number(fillOpacity);
 
     } else {
-      // geoserver does not set a wellKnownName for square explicitly since it is the default value.
-      // Therefore, we have to set the wellKnownName to square if no wellKownName is given.
-      if (!wellKnownName) {
-        _set(sldSymbolizer, 'Graphic[0].Mark[0].WellKnownName[0]._', 'square');
+      if (!fillSymbolizer.color) {
+        fillSymbolizer.opacity = 0;
       }
-      pointSymbolizer = this.getMarkSymbolizerFromSldSymbolizer(sldSymbolizer);
     }
-    return pointSymbolizer;
+
+    if (outlineColor) {
+      fillSymbolizer.outlineColor = outlineColor;
+    }
+    if (outlineWidth) {
+      fillSymbolizer.outlineWidth = Number(outlineWidth);
+    }
+    if (outlineOpacity) {
+      fillSymbolizer.outlineOpacity = Number(outlineOpacity);
+    }
+    if (outlineDashArray) {
+      fillSymbolizer.outlineDasharray = outlineDashArray.split(' ').map(Number);
+    }
+    // TODO: seems like this is missing in the geostyer-stlye
```

Comment:
Could you create an issue for that?

Reply:
https://github.com/geostyler/geostyler-style/issues/500